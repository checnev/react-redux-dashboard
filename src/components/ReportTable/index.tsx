import React, { useState } from 'react';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TablePagination,
  TableHead,
  TableRow,
  Paper,
} from '@material-ui/core';
import type { DailyReport } from 'selectors/stats';

interface ReportTableProps {
  reports: DailyReport[];
}

const useStyles = makeStyles((theme: Theme) => createStyles({
  root: {},
}));

const ReportTable: React.FC<ReportTableProps> = ({ reports }: ReportTableProps) => {
  const [page, setPage] = React.useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(20);

  const handleChangePage = (event: unknown, newPage: number) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  const classes = useStyles();
  return (
    <Paper className={classes.root}>
      <TableContainer>
        <Table className={classes.root} aria-label="car list table">
          <TableHead>
            <TableRow>
              <TableCell>Дата</TableCell>
              <TableCell align="center">Доход</TableCell>
              <TableCell align="center">Кол-во Аренд</TableCell>
              <TableCell align="center">Происшествий</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {reports.length > 0 && reports.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
            .map((report) => (
              <TableRow key={report.date}>
                <TableCell component="th" scope="row">{report.date}</TableCell>
                <TableCell align="center">{report.income}</TableCell>
                <TableCell align="center">{report.countOfLeases}</TableCell>
                <TableCell align="center">{report.incidents}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
      <TablePagination
        rowsPerPageOptions={[20, 50, 100]}
        component="div"
        count={reports.length}
        rowsPerPage={rowsPerPage}
        page={page}
        onChangePage={handleChangePage}
        onChangeRowsPerPage={handleChangeRowsPerPage}
      />
      </Paper>
  );
};

export default ReportTable;
