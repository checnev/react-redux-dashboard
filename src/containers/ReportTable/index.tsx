import { useSelector } from 'react-redux';
import { dailyReportsSelector } from 'selectors/stats';
import ReportTable from 'components/ReportTable';

const ReportTableContainer = () => {
  const reports = useSelector(dailyReportsSelector);

  return <ReportTable reports={reports} />
};

export default ReportTableContainer;
