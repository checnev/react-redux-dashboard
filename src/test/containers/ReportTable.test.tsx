import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import ReportTable from 'containers/ReportTable';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import state from 'test/data/initialState';

const mockStore = configureMockStore([thunk]);

jest.mock('@material-ui/core/TablePagination', () => {
  return function DummyTablePagination() {
    return <div data-testid="pagination"></div>;
  };
});

let container: any = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('ReportTableContainer render', () => {
  const store = mockStore(state);
  act(() => {
    render(
      <Provider store={store}>
        <ReportTable />
      </Provider>,
      container
    );
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
