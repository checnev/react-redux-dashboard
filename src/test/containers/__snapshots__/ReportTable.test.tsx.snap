// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportTableContainer render 1`] = `
"<div class=\\"MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded\\">
  <div class=\\"MuiTableContainer-root\\">
    <table class=\\"MuiTable-root makeStyles-root-1\\" aria-label=\\"car list table\\">
      <thead class=\\"MuiTableHead-root\\">
        <tr class=\\"MuiTableRow-root MuiTableRow-head\\">
          <th class=\\"MuiTableCell-root MuiTableCell-head\\" scope=\\"col\\">Дата</th>
          <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\">Доход</th>
          <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\">Кол-во Аренд</th>
          <th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\">Происшествий</th>
        </tr>
      </thead>
      <tbody class=\\"MuiTableBody-root\\">
        <tr class=\\"MuiTableRow-root\\">
          <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">10.12.2020</th>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">47822</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">23</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">0</td>
        </tr>
        <tr class=\\"MuiTableRow-root\\">
          <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">11.12.2020</th>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">70306</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">25</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">0</td>
        </tr>
        <tr class=\\"MuiTableRow-root\\">
          <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">12.12.2020</th>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">82181</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">27</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">1</td>
        </tr>
        <tr class=\\"MuiTableRow-root\\">
          <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">13.12.2020</th>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">46865</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">29</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">1</td>
        </tr>
        <tr class=\\"MuiTableRow-root\\">
          <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">14.12.2020</th>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">62191</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">31</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">0</td>
        </tr>
        <tr class=\\"MuiTableRow-root\\">
          <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">15.12.2020</th>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">85703</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">33</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">0</td>
        </tr>
        <tr class=\\"MuiTableRow-root\\">
          <th class=\\"MuiTableCell-root MuiTableCell-body\\" role=\\"cell\\" scope=\\"row\\">16.12.2020</th>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">44970</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">35</td>
          <td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter\\">1</td>
        </tr>
      </tbody>
    </table>
  </div>
  <div data-testid=\\"pagination\\"></div>
</div>"
`;
